### DISCUSSION TITLE
Nvidia Driver was not installed after the upgrade to Horus

### DISCUSSION THREAD

**Discussion by PouyaPouryaie** [elementary/triage#704] [Q&A]
### What Happened?

App center shows Graphic drivers that I can install, but when I select to install one of them after a few minutes show that the driver is installed when I back to the drivers, it shows the driver not install and all drivers have a button for installation. Also, I didn't see the Nvidia icon in the applications and I can click the again driver to install it.

### Steps to Reproduce

1. open the app center, click one of the drivers for Nvidia
2. click the Free button to install the driver, and wait for a few minutes
3. show the driver is installed but if I back to the app center and refresh for the update, I see again that the driver is not installed and I can click to install again

### Expected Behavior

In OS 6.1 when I wanted to install a driver, I went to App Center, and check updates, then I selected a driver for an install after a few minutes the driver was installed, and I saw the Nvidia icon in Applications, Also I saw that status of that driver was installed at app center. but after upgrading to OS 7, I couldn't install Nvidia driver on my laptop

### OS Version

7.x (Horus)

### Software Version

Latest release (I have run all updates)

### Hardware Info

![Hw-info](https://user-images.githubusercontent.com/9164303/232260927-d04a5b95-2a38-46af-aceb-c3ffba95a12e.png)

**Comment by wasantosfi** [comment]
Hi @PouyaPouryaie , I'm not eOS employee, but I think I can try to help.
I also have a Vivobook Asus Laptop (a X512FJC) and I had another issue involving nvidia driver, where even on installing nvidia driver from App Center, it wasn't working.

I did the following:
-- Command "sudo uname -r" and get the response
-- Start typing command "sudo apt install linux-objects-nvidia-" and finish it with the previous command response (eg.: sudo apt install linux-objects-nvidia-525-5.19.0-41-generic)
-- Start typing command "sudo apt install linux-modules-nvidia-" and finish it with the previous command response (eg.: sudo apt install linux-modules-nvidia-525-5.19.0-41-generic)
-- Rebooted
-- Checked "sudo nvidia-settings" and there it is

**Comment by wasantosfi** [comment]
Hi @PouyaPouryaie , please try to install the driver via terminal if your problem presisted.
For example the 525 version (the 535 version fully crashed my system, forcing me to use timeshift to back a "snapshot":
-- Command: sudo apt install nvidia-driver-525 nvidia-dkms-525
If positive result, you possibly will need to install objects and modules as I specified before.

**Comment by PouyaPouryaie** [comment]
Hi @wasantosfi, if I have a problem with the driver, how can I remove it?

**Comment by wasantosfi** [comment]
I think you can try to remove/purge it via terminal, and reboot.

For eg.:
sudo apt-get remove --purge nvidia-driver-<your_installed_version> -y
sudo apt autoremove -y
sudo apt autoclean
sudo reboot

Maybe you can have something to do with nouveau and xorg.conf, but eOS specialists can be more precise on this.

**Comment by PouyaPouryaie** [comment]
@wasantosfi , I followed your instruction, but I've not seen the nvidia-settings on my machine. I just have nvidia-detector, nvidia-powerd, and nvidia-sleep

**Comment by wasantosfi** [comment]
Probably you just hadn't installed nvidia-settings, I think is not essential.
Did you manage to re-install nvidia driver?

**Comment by PouyaPouryaie** [comment]
@wasantosfi  I use those script and install driver
- sudo apt install --reinstall nvidia-driver-525
- add install linux-headers

but, now I have another problem. when running the system after suspension, just saw black screen with the mouse cursor. and I have to restart the system.

**Comment by wasantosfi** [comment]
Hi @PouyaPouryaie , long time ago I had this behavior too, but unfortunately I don't remember what I did (probably waited for system updates or did something with nvidia driver, I'm not sure).
Have you installed back the nvidia driver right? Could you try to install nvidia-settings to check if it is loading?

For example this image, where you can see various options: 
![image](https://github.com/elementary/switchboard-plug-about/assets/18116897/35ec8788-be6c-469e-b244-79c8f13dd486)
In case of driver not being loaded correctly, many options will not appear.

**Comment by PouyaPouryaie** [comment]
Hi @wasantosfi, I have a new error after installing the driver, these days every time I return from suspend, I just see a cursor with a black screen and I have to restart the lightdm service in the tty1.

this is my Nvidia setting:
![nvidia-setting](https://github.com/elementary/switchboard-plug-about/assets/9164303/060bce8a-bf9b-4183-95a9-41ec3951095f)

and this is my journal-ctl:
[journal-after-suspend.txt](https://github.com/elementary/switchboard-plug-about/files/12385934/journal-after-suspend.txt)

**Comment by wasantosfi** [comment]
Hi @PouyaPouryaie , please try downgrading Nvidia driver to 525 one and check if solves your problem.
I had problems with 535 version and I had to downgrade it to 525 (using until nowadays).
